packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -117,11 +118,17 @@ export function initWrapperState(element: Element, props: Object) {
   }
 
   var defaultValue = props.defaultValue == null ? '' : props.defaultValue;
+  var initialValue = props.value != null ? props.value : defaultValue;
   var node = ((element: any): InputWithWrapperState);
+
+  if (!shouldSetAttribute('value', initialValue)) {
```

Comment:
Not a blocker, but it might be worth using a method specific to `value` since `shouldSetAttribute` does a [few unnecessary checks in this case](https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/DOMProperty.js#L144-L153). It might also call `shouldAttributeAcceptBooleanValue` [which also does more work than needed](https://github.com/facebook/react/blob/master/packages/react-dom/src/shared/DOMProperty.js#L176-L188)

I don't think this is a really hot path, so it's probably fine the way it is, but just worth noting.

Reply:
Yeah. I had an `assignProperty` method earlier, but we thought it was too much indirection. I'd be happy to revisit this though.

Reply:
Let's fork `shouldSetAttribute` here to `shouldSetValue`? I'm not against a special method per se, I'm against a special method for doing the assignment.